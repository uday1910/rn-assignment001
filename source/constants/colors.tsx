export default {
  red: '#ff0000',
  green: '#3cb371',
  orange: '#ffa500',
  pink: '#ee82ee',
  gray: '#a0a0a0',
  lightGray: '#D3D3D3',
  black: '#404040',
  white: 'white',
  blue: '#3498DB',
  lightBlue: '#D6EAF8',
  lightGreen: '#EAFAF1',
  lightYellow: '#F9E79F',
  lightRed: '#FADBD8',
  extraLightBlue: 'rgba(191, 221, 255,0.4)',
  transparent: 'transparent',
};
