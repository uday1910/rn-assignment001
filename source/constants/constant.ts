import {Platform} from 'react-native';

/**
 * Its Check if device OS is Android
 */
export const IS_ANDROID = Platform.OS === 'android';

/**
 * Its Check if device OS is iOS
 */
export const IS_IOS = Platform.OS === 'ios';
