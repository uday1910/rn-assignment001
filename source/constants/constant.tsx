/**
 * React native imports
 */
import {Dimensions, Platform} from 'react-native';

/**
 * For device window height
 */
export const windowHeight = Dimensions.get('window').height;

/**
 * For device window width
 */
export const windowWidth = Dimensions.get('window').width;

/**
 * Its Check if device OS is Android
 */
export const IS_ANDROID = Platform.OS === 'android';

/**
 * Its Check if device OS is iOS
 */
export const IS_IOS = Platform.OS === 'ios';
