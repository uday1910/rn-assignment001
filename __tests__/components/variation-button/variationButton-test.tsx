import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';

import VariationButton from '../../../source/components/variation-button/variationButton';
import {IS_IOS} from '../../../source/constants/constant';

const mockPressEvent = jest.fn();

const buttonStyle = {
  borderRadius: 10,
  backgroundColor: 'blue',
};

const buttonTextStyle = {
  color: 'white',
  padding: 10,
};

const props = {
  onPressButton: mockPressEvent,
  buttonStyle: buttonStyle,
  activeOpacity: 0.5,
  buttonTextStyle: buttonTextStyle,
  buttonTitle: 'Button',
  addRippleEffect: IS_IOS ? true : false,
};

describe('VariationButton Function', () => {
  it('VariationButton render properly according to the snapshot', () => {
    const button = renderer.create(<VariationButton {...props} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('should have a button', async () => {
    const {getByTestId} = render(<VariationButton {...props} />);
    const button = await getByTestId('variation-button');
    expect(button).toBeDefined();
  });
});
