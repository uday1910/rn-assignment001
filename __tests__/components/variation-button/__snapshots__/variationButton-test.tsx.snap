// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariationButton Function VariationButton render properly according to the snapshot 1`] = `
<View
  accessible={true}
  activeOpacity={0.5}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={
    Object {
      "backgroundColor": "blue",
      "borderRadius": 10,
      "height": 50,
      "width": "85%",
    }
  }
  testID="variation-button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "padding": 10,
        }
      }
    >
      Button
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 0,
        },
      ]
    }
  />
</View>
`;
